.github/workflows/ci.yml

```diff
@@ -0,0 +1,105 @@
+    if: >-
+        !contains(github.event.head_commit.message, '[skip ci]')
+        && !contains(github.event.head_commit.message, '[skip tests]')
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+        include:
+          - python-version: "3.7"
+          - python-version: "3.7"
+            extra-env: "PIP_EXTRA_INSTALLATION='numpy symengine==0.5.0' USE_SYMENGINE=1"
+          - python-version: "3.7"
+            extra-env: "PIP_EXTRA_INSTALLATION='sympy==1.4'"
+          - python-version: "3.8"
+            extra-env: "PIP_EXTRA_INSTALLATION='https://github.com/sympy/sympy/archive/master.zip'"
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install -r test_requirements.txt
+        python -m pip install -e .
+        if [ -n "$PIP_EXTRA_INSTALLATION" ]; then
+          # deliberately not quoted
+          python -m pip install $PIP_EXTRA_INSTALLATION
+        fi
+        python -m pip check
+    - name: Lint
+      run: |
+        python -m pip install flake8 flake8_formatter_junit_xml
+        mkdir -p test-reports/flake8
+        flake8 -v
+    - name: Test
+      run: |
+        PYTEST_ARGS=();
+        PYTEST_ARGS+=(-n 2 --dist loadscope);
+        if [[ ! -z "$PYTEST_K_FILTER" ]]; then
+          PYTEST_ARGS+=(-k "$PYTEST_K_FILTER");
+        fi;
+        mkdir -p test-reports/pytest
+        pytest \
+          --cov=galgebra \
+          --nbval examples/ipython/ \
+          test \
+          --current-env \
+          --sanitize-with test/.nbval_sanitize.cfg \
+          --junitxml=test-reports/pytest/results.xml \
+          "${PYTEST_ARGS[@]}"
+    - name: Upload coverage to Codecov
+      if: "matrix.python-version == '3.11'"
+      uses: codecov/codecov-action@v3
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+
+  release:
+    name: Create release and send to PyPI
+    needs: build
+    if: >-
+        github.ref_type == 'tag'
+        && startsWith(github.ref, 'refs/tags/v')
+        && !contains(github.event.head_commit.message, '[skip ci]')
+        && !contains(github.event.head_commit.message, '[no release]')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Install dependencies
+        run: |
+          python -m pip install -r requirements.txt
+      - name: Build and upload package
+        if: success()
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
+        run: |
+          python setup.py sdist bdist_wheel
+          twine upload dist/*
```

Comment:
I think I'd prefer to use https://github.com/marketplace/actions/pypi-publish here

Reply:
I agree and it uses twine under the hood

Reply:
It can be done in a separate, smaller follow-up PR. Just checked that my CircleCI script was also using twine, it just has ported here.